### PR TITLE
do not override already set content-type, fixes #1120

### LINKED ISSUE #1120: Koa always overwrite `Content-Type` for JSON responses
```js
const Koa = require('koa');
const app = new Koa();
app.use(ctx => {
  ctx.set('Content-Type', 'application/vnd.myapi.v1+json');
  ctx.body = {message: 'hello'};
});
app.listen(3000);
```

When running this code I aways get `Content-Type: application/json; charset=utf-8`.

I can make it work if I write the header after setting the body. It only affects `json` responses. I'm not sure if that's a bug, but I couldn't find any information about that behavior in documentation.

### PR DESCRIPTION
**Pull request by fl0w**
semver major, #1114

This could be done as semver minor - if an application setting is addd, e.g. `app.responsOverride = true` or something.

### CODE REVIEW

**Comment by jonathanong** [comment]
this might break apps if someone is relying on this behavior, so i would wait for semver major.

**Comment by jonathanong** [comment]
unless we only set it to json if the type is not json-like

**Comment by fl0w** [comment]
Yes, definitely semver-major.
Though with this PR, #1137, #1115 and a few more a v3 is already justified (IMO).

It would be nice to drop a v3 alpha, call out EOL for v1 (soon-ish?) and break the stigma of having to rewrite Koa to justify a major bump and add proper deprecations to v2 and v1.

**Comment by niftylettuce** [comment]
I agree @fl0w

**Comment by jkomyno** [comment]
Any news about this PR? @jonathanong

**Review by 3imed-jaberi — APPROVED** [review]

**Comment by jkomyno** [comment]
Hi, it's been two years, any update on this PR?
@jonathanong @3imed-jaberi

File: test/response/body.js [hunk 1]
```diff
@@ -15,14 +15,14 @@ describe('res.body=', () => {
...
-      it('should override as json', () => {
+      it('should not override as json', () => {
         const res = response();
 
         res.body = '<em>hey</em>';
         assert.equal('text/html; charset=utf-8', res.header['content-type']);
 
         res.body = { foo: 'bar' };
-        assert.equal('application/json; charset=utf-8', res.header['content-type']);
+        assert.equal('text/html; charset=utf-8', res.header['content-type']);
```
Comment (jonathanong) [inline test/response/body.js:25]: this test case is already updated, so this feature must already be live

**Pull request closed without merging**